### DISCUSSION TITLE
Are Zustand devtools deprecated or not?

### DISCUSSION THREAD

**Discussion by coltanium13** [pmndrs/zustand#1437] [Q&A] [answered]
In zustand v3, i have deprecation warnings. However, i am also seeing change logs in the zustand v4 docs for using devtools. 

Should i stop using devtools in v3, or stop using it?

Is zustand v4 going to allow and support devTools, or is it officially gone? 

The docs still say you can use devtools. So I'm confused. Thank you!

**Comment by dai-shi** [comment, marked as answer]
No, the devtools middleware is fully supported in v4. There's also a WIP PR #1435 to improve it.

In the latest v3, some parameters of `devtools` are deprecated. Once you resolve it, it should just work in v4 too. Or just try to upgrade to v4 and see how it works.

**Reply by coltanium13** [reply]
Ok, it vs code is saying devtools itself is deprecated in the version im using. It probably needs to be updated to the lates v3. I have also noticed that the version im using, the devtools, enabled parameter, does not respect being turned to false, when testing, and the test always logs a waring that redux dev tools needs to be installed.

I will try to update to v4 as soon as we can, and hopefully it just fixes all the issues. Its still using redux devtools as the extension right?

thank you for the fast reply!

**Reply by dai-shi** [reply]
> it vs code is saying devtools itself is deprecated in the version im using.

Oh, I see. That's not our intention. Only some parameters are deprecated. Maybe, it can't be helped with `@deprecation` behavior.

**Comment by coltanium13** [comment]
Ah ok.

Did the enabled parameter get fixed? So if you make it false if not env ==
development it won't show the console warning while running tests?

I saw that was an issues and they said it was fixed, but it didn't seem to
work. Maybe it works in v4?

Thank you!

-Colton

**Reply by dai-shi** [reply]
I believe so.
